dask/array/tests/test_array_core.py

```diff
@@ -3611,12 +3611,61 @@ def test_setitem_2d():
     assert_eq(x, dx)
 
 
+def test_setitem_extended_API():
+    x = np.arange(60).reshape((6, 10))
+    dx = da.from_array(x.copy(), chunks=(2, 2))
+
+    x[::2, ::-1] = -1
+    x[1::2] = -2
+    x[:, [3, 5, 6]] = -3
+    x[[True, False, False, False, False, False], 2] = -4
+    x[2:4, x[0] > 3] = -5
+    x[2, x[0] < -2] = -7
+    x[x % 2 == 0] = -8
+    x[[4, 3, 1]] = -9
+    x[5, ...] = -10
+    x[2:4, 5:1:-2] = -x[:2, :2]
+    x[:2, :3] = [[1, 2, 3]]
+
+    dx[::2, ::-1] = -1
+    dx[1::2] = -2
+    dx[:, [3, 5, 6]] = -3
+    dx[[True, False, False, False, False, False], 2] = -4
+    dx[2:4, dx[0] > 3] = -5
+    dx[2, dx[0] < -2] = -7
+    dx[dx % 2 == 0] = -8
+    dx[[4, 3, 1]] = -9
+    dx[5, ...] = -10
+    dx[2:4, 5:1:-2] = -dx[:2, :2]
+    dx[:2, :3] = [[1, 2, 3]]
+
+    assert_eq(x, dx)
+
+
 def test_setitem_errs():
     x = da.ones((4, 4), chunks=(2, 2))
 
     with pytest.raises(ValueError):
         x[x > 1] = x
 
+    # Multiple 1-d boolean/integer arrays
+    with pytest.raises(ValueError):
+        x[[1, 2], [2, 3]] = x
+
+    with pytest.raises(ValueError):
+        x[[True, True, False, False], [2, 3]] = x
+
+    with pytest.raises(ValueError):
+        x[[True, True, False, False], [False, True, False, False]] = x
+
+    # Not strictly monotonic
+    with pytest.raises(ValueError):
+        x[[1, 3, 2]] = x
```

Comment:
This sort of indexing is supported in NumPy, so it would be better to raise a different, more informative error (`NotImplemetedError`?) to clarify that this currently isn't supported in dask but might work eventually.

Reply:
Good idea.